hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/AppContext.java

```diff
@@ -68,4 +70,10 @@ interface Gossip {
      * @return the gossip interface
      */
     Gossip gossip();
+
+    /**
+     * The metrics provider for the system
+     * @return the metrics provider
+     */
+    Supplier<Metrics> metricsSupplier();
```

Comment:
We do our best to keep service-specific information in its corresponding service. Since contract metrics are only intended for the contract service, I'm thinking we create a `ContractContext` interface, similar to `TokenContext`, that makes these metrics available _only_ to SCSC. Will continue to think on this.